src/openrct2/core/String.cpp

```diff
@@ -343,66 +324,35 @@ namespace String
         return buffer;
     }
 
-    utf8* Format(const utf8* format, ...)
+    std::string StdFormat(const utf8* format, ...)
     {
         va_list args;
         va_start(args, format);
-        utf8* result = Format_VA(format, args);
+        std::string result = Format_VA(format, args);
         va_end(args);
         return result;
     }
 
-    utf8* Format_VA(const utf8* format, va_list args)
+    u8string Format_VA(const utf8* format, va_list args)
     {
-        va_list args1, args2;
-        va_copy(args1, args);
-        va_copy(args2, args);
-
-        // Try to format to a initial buffer, enlarge if not big enough
-        size_t bufferSize = 4096;
-        utf8* buffer = Memory::Allocate<utf8>(bufferSize);
+        va_list copy;
 
-        // Start with initial buffer
-        int32_t len = vsnprintf(buffer, bufferSize, format, args);
-        if (len < 0)
-        {
-            Memory::Free(buffer);
-            va_end(args1);
-            va_end(args2);
+        va_copy(copy, args);
 
-            // An error occurred...
-            return nullptr;
-        }
+        // Try to format to a initial buffer, enlarge if not big enough
+        int32_t len = vsnprintf(nullptr, 0, format, copy);
 
-        size_t requiredSize = static_cast<size_t>(len) + 1;
-        if (requiredSize > bufferSize)
-        {
-            // Try again with bigger buffer
-            buffer = Memory::Reallocate<utf8>(buffer, bufferSize);
-            len = vsnprintf(buffer, bufferSize, format, args);
-            if (len < 0)
-            {
-                Memory::Free(buffer);
-                va_end(args1);
-                va_end(args2);
+        va_end(copy);
 
-                // An error occurred...
-                return nullptr;
-            }
-        }
-        else
+        if (len >= 0)
         {
-            // Reduce buffer size to only what was required
-            bufferSize = requiredSize;
-            buffer = Memory::Reallocate<utf8>(buffer, bufferSize);
+            std::string buffer(std::size_t(len) + 1, '\0');
+            vsnprintf(buffer.data(), buffer.size(), format, args);
+            buffer.resize(len);
+            return buffer;
         }
 
-        // Ensure buffer is terminated
-        buffer[bufferSize - 1] = '\0';
-
-        va_end(args1);
-        va_end(args2);
-        return buffer;
+        return std::string{};
```

Comment:
For completeness sake, add a `log_warning` here to show that something is amiss, instead of failing silently.

Reply:
Added "Inputted buffer length is less than 0" warning